torch/testing/_internal/common_methods_invocations.py

```diff
@@ -88,28 +118,68 @@ class UnaryUfuncInfo(OpInfo):
                  dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
                  dtypesIfCUDA=floating_and_complex_types_and(torch.half),
-                 dtypesIfROCM=floating_types_and(torch.half, torch.bfloat16),
+                 dtypesIfROCM=floating_types_and(torch.half),
+                 domain=(None, None),  # the [low, high) domain of the function
+                 handles_large_floats=True,  # whether the op correctly handles large float values (like 1e-20)
+                 handles_extremals=True,  # whether the op correctly handles extremal values (like inf)
+                 handles_complex_extremals=True,  # whether the op correct handles complex extremals (like inf -infj)
                  **kwargs):
         super(UnaryUfuncInfo, self).__init__(name,
                                              dtypes=dtypes,
                                              dtypesIfCPU=dtypesIfCPU,
                                              dtypesIfCUDA=dtypesIfCUDA,
                                              dtypesIfROCM=dtypesIfROCM,
                                              **kwargs)
+        self.ref = ref
+        self.domain = domain
+        self.handles_large_floats = handles_large_floats
+        self.handles_extremals = handles_extremals
+        self.handles_complex_extremals = handles_complex_extremals
 
 L = 20
 M = 10
 S = 5
 
+# Situational skips for tests that fail on particular builds
+_windows_skip = ('.*numerics.*complex.*',) if IS_WINDOWS else tuple()
+_mac_skip = ('.*numerics.*cpu.*complex.*',) if IS_MACOS else tuple()
+_rocm_skip = ('.*numerics.*float32.*',) if TEST_WITH_ROCM else tuple()
+
 # Operator database
 op_db = [
+    # NOTE: CPU complex acos produces incorrect outputs (https://github.com/pytorch/pytorch/issues/42952)
+    UnaryUfuncInfo('acos',
+                   ref=np.arccos,
+                   domain=(-1, 1),
+                   handles_complex_extremals=False,
+                   decorators=(precisionOverride({torch.bfloat16: 1e-1,
+                                                  torch.complex64: 1e-2}),),
+                   tests_to_skip=('.*numerics.*cpu.*complex.*',) + _windows_skip),
     UnaryUfuncInfo('cos',
-                   dtypesIfCUDA=floating_and_complex_types_and(torch.half, torch.bfloat16)),
+                   ref=np.cos,
+                   dtypesIfCUDA=floating_and_complex_types_and(torch.half, torch.bfloat16),
+                   handles_large_floats=False,
+                   decorators=(precisionOverride({torch.bfloat16: 1e-2}),),
+                   tests_to_skip=_windows_skip + _mac_skip + _rocm_skip),
     UnaryUfuncInfo('cosh',
-                   dtypesIfCPU=floating_and_complex_types()),
+                   ref=np.cosh,
+                   dtypesIfCPU=floating_and_complex_types(),
+                   tests_to_skip=_windows_skip + _mac_skip),
+    UnaryUfuncInfo('sin',
+                   ref=np.sin,
+                   handles_large_floats=False,
```

Comment:
This is kind of surprising, because I would have thought at least some of our libraries use infinite pi reduction and would therefore "handle" large floats equivalently (albeit in a dubious way: assuming that the input was infinitely precisely represented, which is kind of a goofy thing to assume, but a pretty useful thing if you're going for bit-for-bit reproducibility)

Reply:
This and cosine failing to pass the tests with large float values will require additional investigation. In particular, I don't think our different toolchains across operating systems produce consistent values for these functions.